rpmvenv/extensions/loader.py

```diff
@@ -21,10 +21,11 @@ class InvalidDependency(Exception):
 
 def load_extensions(whitelist=()):
     """Get an iterable of extensions in order."""
-    whitelist = tuple(set(('core',) + tuple(whitelist)))
+    whitelist = tuple(('core',) + tuple(whitelist))
+    unique_whitelist = tuple(set(whitelist))
```

Comment:
The change in behavior comes from not overwriting the original whitelist. We only use unique_whitelist (the one passed through set()), when creating extensions. Instead of sorting on the unique_whitelist (which is what was happening before), we sort on the original whitelist that has order preserved (wasn't passed through set()).